### ISSUE TITLE
UserSession.getTokenForServer constructs invalid url for rest/info

### ISSUE THREAD

**Issue by kenlyon** [Esri/arcgis-rest-js#581]
Using Node.js version 10.15.3, here is what happens when using `request` with credentials when discovering the services offered by a server.

## Example 1

```
const session = new UserSession({
    username: "<myUsername>",
    password: "<myPassword>",
});

const data = await request("https://<hostname>/arcgis/rest/services",
                           { authentication: session });
```
* `UserSession.getServerRootUrl` returns `"https://<hostname>/arcgis/rest/services"`
* `UserSession.getTokenForServer` tries to contact
`https://<hostname>/arcgis/rest/services/rest/info` instead of  
`https://<hostname>/arcgis/rest/info`.

## Example 2

```
const session = new UserSession({
    username: "<myUsername>",
    password: "<myPassword>",
});

const data = await request("https://<hostname>/arcgis/rest/",
                           { authentication: session });
```
* `UserSession.getServerRootUrl` returns `"https://<hostname>/arcgis/rest/"`
* `UserSession.getTokenForServer` tries to contact
`https://<hostname>/arcgis/rest/rest/info` instead of  
`https://<hostname>/arcgis/rest/info`.

**Comment by jgravois** [comment]
hi @kenlyon :wave:

the current regex tests for `services/` with a trailing slash.

https://github.com/Esri/arcgis-rest-js/blob/4d2975a0392488afff414f64784db12c0eea081c/packages/arcgis-rest-auth/src/UserSession.ts#L784

**Comment by kenlyon** [comment]
Ah, I see. Right enough, when I added the slash to my url it behaved correctly. I guess you weren't expecting anyone to hit the services endpoint directly. 😊

Here's a suggested change to that regex:
`/\/rest(\/admin)?\/services(?:\/|#|\?|$)`

* I included all delimiters after "services" that I could think of. (slash, pound, question mark and end of string)
* I used a non-capturing group to avoid adding anything extra to the result, although I noticed the first group is capturing.

Either way, I'll keep the slash on the end of my url for now.

Thanks for the quick response!
